### PR TITLE
EVG-13466 task annotations put does not add the source to Issues

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: model/annotations/task_annotations.go [hunk 1]
```diff
@@ -162,15 +162,15 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 		a.Note.Source = source
 		update[NoteKey] = a.Note
 	}
-	if a.Issues != nil {
-		for _, issue := range a.Issues {
-			issue.Source = source
+	if len(a.Issues) != 0 {
```
Comment (ablack12) [inline model/annotations/task_annotations.go:165]: Oh, I actually don't think we should do this. If there's currently an Issue in issue, how else do I indicate I want to remove it and replace it with nothing? I assume if I give a document `{"task_id": 123, "issues": []} ` issues would be not nil but length 0, and suspected_issues would be nil. Is this not the case?

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: model/annotations/task_annotations.go (hunk 1 above)
Comment (malikchaya2) [reply model/annotations/task_annotations.go:165]: If issues is not provided, it it not nil, it is an empty array, so checking for nil is not an option. I hear the concern though, I'll think about it.

Comment (malikchaya2) [reply model/annotations/task_annotations.go:165]: In your example, suspected_issues would be `<[]github.com/evergreen-ci/evergreen/model/annotations.IssueLink> (length: 0, cap: 0)`

Comment (ablack12) [reply model/annotations/task_annotations.go:165]: This seems like a failure of https://github.com/evergreen-ci/evergreen/blob/39b9a4171b64ba3afea10685c549e5f02b2dc197/rest/model/task_annotations.go#L145; you could change this to return nil if t is nil? Assuming t comes in as nil for suspected_ssues in that example.

Comment (malikchaya2) [reply model/annotations/task_annotations.go]: thanks!

**Review by ablack12 — APPROVED** [review]

File: rest/route/annotations_test.go [hunk 2]
```diff
@@ -465,7 +475,7 @@ func TestAnnotationByTaskPutHandlerRun(t *testing.T) {
 	assert.Equal(t, "task-1-note_0", annotation.Note.Message)
 	assert.Equal(t, "test_annotation_user", annotation.Note.Source.Author)
 	assert.Equal(t, "api", annotation.Note.Source.Requester)
-
+	assert.Equal(t, "api", annotation.Issues[0].Source.Requester)
```
Comment (ablack12) [inline rest/route/annotations_test.go:478]: Can you add something to the "test update" below to verify that if Issues/SuspectedIssues isn't defined then it won't be updated?

Comment (ablack12) [reply rest/route/annotations_test.go:478]: I don't need to see it again for you to merge though.

**Pull request merged**